packages/rest/src/http.ts

```diff
@@ -62,6 +62,23 @@ export function hasBody(program: Program, parameters: ModelTypeProperty[]): bool
   return parameters.find((p) => isBody(program, p)) !== undefined;
 }
 
+const statusCodeKey = Symbol();
+export function $statusCode(program: Program, entity: Type) {
+  if (entity.kind === "ModelProperty") {
+    program.stateSet(statusCodeKey).add(entity);
+  } else {
+    reportDiagnostic(program, {
+      code: "decorator-wrong-type",
+      format: { decorator: "statusCode", entityKind: entity.kind },
+      target: entity,
+    });
+  }
+}
+
+export function isStatusCode(program: Program, entity: Type) {
```

Comment:
I am curious why an 'isStatusCode' method for the property type containing a status code, rather than a 'getStatusCode' method that would return the status code for a model type?  To me, it feels like the http library should define what a status code is (numeric value, set of numeric values, etc. and make this an intrinsic definition of http).  This would allow the http library to interpret values like 'OK' or '200' as equivalent without extensions having to do this, and perhaps doing it inconsistently.

Reply:
I would _love_ to define all the status codes by name in the http library.  I even coded that up as a enum (which made the most sense to me).  Unfortunately, the Cadl language currently does not allow references to enum instances.  When we get that support, then we certainly should add this and encourage the use of `http.Ok` over `200`.

As for `isStatusCode` vs `getStatusCode`, I think you are really asking whether we want to decorate a property with the status code value or make the decorator attach to a model with the status code value as a parameter. I don't see any real downside to attaching the decorator to a model, and it would be a little cleaner in that it eliminates the property name which was simply ignored.

Reply:
We've decided to leave `@statusCode` as a decorator on a property with no restriction on the property name. But libraries built on top of Cadl / the http library could impose a naming convention if desired.

Reply:
This is more about statusCode values (so a required type for the ModelTypeProperty), but you were going to have a follow-up that enabled multiple values for status code, so I will defe